### PR TITLE
Add Tools section in Engine Developer Guide

### PR DESCRIPTION
**Pull request by willihay**
Signed-off-by: Willow Hayward <17654918+willihay@users.noreply.github.com>

## Change summary

- Add Tools section in Engine Developer Guide, including an index entry in the Guide overview, and a stub for Project Manager.
- Fix broken links in other overviews.
- Remove unnecessary weight from Script Canvas Gem topic.

https://deploy-preview-2177--o3deorg.netlify.app/docs/engine-dev/tools/

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

**Comment by thefranke** [comment]
So this PR is to add the stubs for these topics? Is there already someone writing the individual sections?

File: content/docs/engine-dev/tools/project-manager/_index.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+linkTitle: Project Manager
+title: Project Manager Developer Documentation
+description: Developer documentation for the Project Manager tool.
+---
+
+{{< note >}}
+This information is for developers of the **Project Manager** tool. If you're a user working with Project Manager, please refer to the [Project Manager User Guide](/docs/user-guide/project-config/project-manager).
```
Comment (tonybalandiuk) [inline content/docs/engine-dev/tools/project-manager/_index.md:8]: nit - do we typically use conjunctions in O3DE documentation?  Unsure, just checking.

**Review by tonybalandiuk — APPROVED** [review]
I approve these changes. Adds clarity to the documentation. Left 1 nit comment, which should not prevent this submission.

**Comment by willihay** [comment]
> So this PR is to add the stubs for these topics? Is there already someone writing the individual sections?

This PR is mostly for adding a Tools section for the upcoming Project Manager developer docs, but it also fixes some earlier stubs.

There is someone working on the Project Manager docs, but the other stubs were created as placeholders for the benefit of technical SIGs (in accordance with the Engine Developer Guide RFC) in anticipation of contributors picking up sections there.

File: content/docs/engine-dev/tools/project-manager/_index.md (hunk 1 above)
Comment (willihay) [reply content/docs/engine-dev/tools/project-manager/_index.md:8]: Yes, it was encouraged by AWS editors.

**Review by Ulrick28 — APPROVED** [review]
Approved as this work is the expected outcome from earlier conversations, with the Project Manager being the test case.

**Pull request merged**